From 8eba88632323f6d65e5e68a2830a9ef60dd09d91 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mart=C3=ADn=20Lucas=20Golini?= <spartanj@gmail.com>
Date: Sun, 10 Aug 2025 11:52:46 -0300
Subject: [PATCH] Allow to build without X11 on Linux and BSD. Added flags:
 `SOIL_EGL` flag to indicate the build have access to an EGL context.
 `SOIL_NO_X11` flag to force disable using X11 features.

---
 src/SOIL2/SOIL2.c | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/src/libprojectM/Renderer/SOIL2/SOIL2.c b/src/libprojectM/Renderer/SOIL2/SOIL2.c
index 42f111a..d220e85 100644
--- a/src/libprojectM/Renderer/SOIL2/SOIL2.c
+++ b/src/libprojectM/Renderer/SOIL2/SOIL2.c
@@ -29,7 +29,11 @@
 	#endif
 #elif defined( __ANDROID__ ) || defined( ANDROID )
 	#define SOIL_PLATFORM_ANDROID
-#elif ( defined ( linux ) || defined( __linux__ ) || defined( __FreeBSD__ ) || defined(__OpenBSD__) || defined( __NetBSD__ ) || defined( __DragonFly__ ) || defined( __SVR4 ) )
+#elif !defined( SOIL_GLES2 ) && !defined( SOIL_GLES1 ) && !defined( SOIL_NO_X11 ) && \
+	!defined( SOIL_EGL ) &&                                                          \
+	( defined( linux ) || defined( __linux__ ) || defined( __FreeBSD__ ) ||          \
+	  defined( __OpenBSD__ ) || defined( __NetBSD__ ) || defined( __DragonFly__ ) || \
+	  defined( __SVR4 ) )
 	#define SOIL_X11_PLATFORM
 #endif
 
@@ -37,8 +41,9 @@
 	#define SOIL_GLES2
 #endif
 
-#if ( defined( SOIL_GLES2 ) || defined( SOIL_GLES1 ) ) && !defined( SOIL_NO_EGL ) && !defined( SOIL_PLATFORM_IOS )
-	#include <EGL/egl.h>
+#if ( defined( SOIL_GLES2 ) || defined( SOIL_GLES1 ) || defined( SOIL_EGL ) ) && \
+	!defined( SOIL_NO_EGL ) && !defined( SOIL_PLATFORM_IOS )
+#include <EGL/egl.h>
 #endif
 
 #if defined( SOIL_GLES2 )
@@ -216,7 +221,7 @@
 
 #if defined( SOIL_PLATFORM_IOS )
 	func = dlsym( RTLD_DEFAULT, proc );
-#elif defined( SOIL_GLES2 ) || defined( SOIL_GLES1 )
+#elif defined( SOIL_GLES2 ) || defined( SOIL_GLES1 ) || defined( SOIL_EGL )
 	#ifndef SOIL_NO_EGL
 		func = eglGetProcAddress( proc );
 	#else
@@ -2973,7 +2978,7 @@
 				ext_addr = (P_SOIL_GLGENERATEMIPMAPPROC)SOIL_GL_GetProcAddress("glGenerateMipmapEXT");
 			}
 
-			#elif defined( SOIL_GLES2 )
+			#elif defined( SOIL_GLES2 ) || defined( SOIL_EGL )
 				ext_addr = 	&glGenerateMipmap;
 			#else /** SOIL_GLES1 */
 				ext_addr = &glGenerateMipmapOES;
